homeassistant/components/pvpc_hourly_pricing/config_flow.py

```diff
@@ -0,0 +1,65 @@
+"""Config flow for pvpc_hourly_pricing."""
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import callback
+
+from . import CONF_NAME, UI_CONFIG_SCHEMA
+from .const import ATTR_TARIFF, DEFAULT_TARIFF, DOMAIN, TARIFFS
+
+DOMAIN_NAME = DOMAIN
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle the Options flow for `pvpc_hourly_pricing` to change the tariff."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry):
+        """Initialize the options flow handler with the config entry to modify."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Manage the options."""
+        if user_input is not None:
+            return self.async_create_entry(
+                title=self.config_entry.data.get(CONF_NAME), data=user_input,
+            )
+
+        current_tariff = self.config_entry.options.get(
+            ATTR_TARIFF, self.config_entry.data.get(ATTR_TARIFF, DEFAULT_TARIFF)
+        )
+        return self.async_show_form(
+            step_id="init",
+            data_schema=vol.Schema(
+                {vol.Required(ATTR_TARIFF, default=current_tariff): vol.In(TARIFFS)}
+            ),
+        )
+
+
+class TariffSelectorConfigFlow(config_entries.ConfigFlow, domain=DOMAIN_NAME):
+    """Handle a config flow for `pvpc_hourly_pricing` to select the tariff."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        if user_input is not None:
+            if any(
+                user_input[CONF_NAME] == entry.data[CONF_NAME]
```

Comment:
This needs a unique id here.
```
await self.async_set_unique_id(YOUR_UNIQUE_ID_HERE)
self._abort_if_unique_id_configured()
```
It looks like the tariff is the unique id here, but you'll have to remove the options flow (this seems like its fine though since they can just remove/add if they change tariffs)

Reply:
<img width="568" alt="Screen Shot 2020-03-19 at 9 54 14 AM" src="https://user-images.githubusercontent.com/663432/77080905-e33c2700-69c7-11ea-9075-fec47d8adf45.png">

Reply:
> It looks like the tariff is the unique id here, but you'll have to remove the options flow (this seems like its fine though since they can just remove/add if they change tariffs)

I thought about it when I did the implementation. 

I wanted to give it a try to the OptionsFlow (1st time using it), as I think more config options will come to this integration (like contracted power and things alike, to give more than just the un-taxed energy cost). But for this initial proposal, the only excuse to use it was to make the tariff selection changeable in options :)

In that context, as the tariff can be changed, I excluded it from unique_id and added the entity_id.

I didn't think about the entity_id possible renaming in UI, my bad. It is true that this makes it out of `unique_id`, and the only thing left is the tariff.

**So... removing all OptionsFlow, right? :(**
That also limits the number of sensors that can be created, to just 3, one per tariff.

Reply:
How about removing in the initial merge adding it back in a followup PR when you add more options?

Reply:
yeah, let's do that.

BTW, there are also other things that I was not sure about them, like the multiple ways of configuring the sensor. I'm not sure if any of them are deprecated. It can be configured:
- With integrations UI
- As a sensor platform, `sensor: - platform: pvpc_hourly_pricing`
- As an integration, `pvpc_hourly_pricing: - tariff: xxx`

And there are tests for all three, but not sure if it is good to include all.

Reply:
Since you are using a config flow, I think it makes the most sense to just do it as an integration in the UI and as an integration in the yaml  & remove the sensor platform option.

Reply:
Ok, I'll remove everything and push the changes. That will make this PR considerably smaller :)

Reply:
@bdraco, all done in the last commit. Also changed the docs in https://github.com/home-assistant/home-assistant.io/pull/12194/commits/10084e462dca50b053004b02abce317da1dd4dd3